core/src/main/java/org/apache/carbondata/core/datamap/dev/DataMap.java

```diff
@@ -67,4 +68,14 @@
    */
   void clear();
 
+  /**
+   * This method will be required for dataMaps that require 2 stage of construction.
+   * Ideal scenario will be first stage contains all the processing logic and second
+   * stage includes updating to database.
+   * Method usage can differ based on scenario and implementation
+   *
+   * @throws MemoryException
+   */
+  void commit() throws MemoryException, IOException;
```

Comment:
I think it does not make sense to have this method in interface level. Caching should only be restricted to DataMapFactory not to the level of DataMap.